tests/eval_files/async912.py

```diff
@@ -0,0 +1,108 @@
+# ASYNCIO_NO_ERROR
+import trio
+
+
+async def foo():
+    with trio.move_on_after(0.1):  # error: 4
+        ...
+    with trio.move_on_at(0.1):  # error: 4
+        ...
+    with trio.fail_after(0.1):  # error: 4
+        ...
+    with trio.fail_at(0.1):  # error: 4
+        ...
+    with trio.CancelScope(0.1):  # error: 4
+        ...
```

Comment:
Feels like we should just cover the `CancelScope` case with `ASYNC100` and reserve this optional check for the cases where the await is conditional.

It seems like a semantically distinct problem: "this timeout can't fire" vs "you might be unable to cancel this code like you expected to"... but Trio says that no, these are actually the same thing.  We should be really clear about that in the docs.

Reply:
This turned out to be real tricky to fix properly - but ASYNC100 is now fully handled by Visitor91x so as not to have its logic duplicated. That surfaced a bunch of issues, but those are all handled.